src/pal/inc/pal.h

```diff
@@ -201,7 +201,13 @@ extern "C" {
 
 #else // _MSC_VER
 
+#if defined (_M_ARM) && !defined (DEBUG)
```

Comment:
Please use `_ARM_` instead of `_M_ARM`.
And why do you define it differently for DEBUG and non-DEBUG? It seems it would not hurt anything.